app/src/main/java/com/example/lwilcox/lab2/SearchFragment.java

```diff
@@ -0,0 +1,127 @@
+    public Integer photoIndex;
+    public ArrayList<String> imageLinks;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        myFragmentView = inflater.inflate(R.layout.fragment_search, container, false);
+
+        searchButton = (Button) myFragmentView.findViewById(R.id.search);
+        nextButton = (Button) myFragmentView.findViewById(R.id.next);
+        previousButton = (Button) myFragmentView.findViewById(R.id.previous);
+        saveButton = (Button) myFragmentView.findViewById(R.id.save);
+        viewStreamButton = (Button) myFragmentView.findViewById(R.id.viewStream);
+        editText = (EditText) myFragmentView.findViewById(R.id.editText);
+        imageView = (ImageView) myFragmentView.findViewById(R.id.imageView);
+        final FeedReaderDbHelper FeedReaderDBH = new FeedReaderDbHelper(getActivity().getApplicationContext());
+
+        searchButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String searchText = editText.getText().toString();
+                HttpHandler handler = new HttpHandler(getActivity().getApplicationContext());
+                photoIndex = 0;
+                handler.imageSearch(searchText, new Callback() {
+                    @Override
+                    public void callback(ArrayList<String> displayList) {
+                        imageLinks = displayList;
+                        setImage();
+                    }
+                });
+            }
+        });
+        nextButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                try {
+                    if (photoIndex < (imageLinks.size() - 1)) {
+                        photoIndex ++;
+                    }
+                    else {
+                        photoIndex = 0;
+                    }
+                    setImage();
+                } catch (Exception e) {
+                    Context context = getActivity().getApplicationContext();
+                    CharSequence text = context.getResources().getString(R.string.search_first);
+                    int duration = Toast.LENGTH_SHORT;
+                    Toast toast = Toast.makeText(context, text, duration);
+                    toast.show();
+                }
+            }
+        });
+        previousButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                try {
+                    if (photoIndex > 0) {
+                        photoIndex --;
+                    }
+                    else {
+                        photoIndex = (imageLinks.size() - 1);
+                    }
+                    setImage();
+                } catch (Exception e) {
+                    Context context = getActivity().getApplicationContext();
```

Comment:
This code block is duplicated a lot. It would be cleaner if you wrapped this in a method and then called the method in each of these spots.